cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java

```diff
@@ -51,12 +68,82 @@ public boolean equals(Object o) {
         return getType().equals(that.getType())
-                && getIdentity().equals(that.getIdentity())
-                && getAffiliation().equals(that.getAffiliation());
+               && getIdentity().equals(that.getIdentity())
+               && getAffiliation().equals(that.getAffiliation());
     }
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(getType(), getIdentity(), getAffiliation());
+    public CristinPerson toCristinPerson() {
+        CristinPerson cristinPerson = new CristinPerson();
+        cristinPerson.setCristinPersonId(extractLastPathElement(this.getIdentity().getId()));
+        cristinPerson.setUrl(this.getIdentity().getId().toString());
+        cristinPerson.setFirstName(this.getIdentity().getFirstName());
+        cristinPerson.setSurname(this.getIdentity().getLastName());
+        cristinPerson.setRoles(extractCristinRoles());
+        return cristinPerson;
+    }
+
+    private List<CristinRole> extractCristinRoles() {
+        return contributorTypeCanBeMappedToCristinRole()
+                   ? getCristinRoles()
+                   : Collections.emptyList();
+    }
+
+    private boolean contributorTypeCanBeMappedToCristinRole() {
+        return getCristinRole(getType()).isPresent();
+    }
+
+    private List<CristinRole> extractCristinRoles(NvaContributor contributor) {
+        return contributorTypeCanBeMappedToCristinRole()
+                   ? getCristinRoles()
+                   : Collections.emptyList();
+    }
+
+    private List<CristinRole> getCristinRoles() {
+        var cristinRole = new CristinRole();
+        addRolesBasedOnContributorsType(cristinRole);
+        addOrganizationInformationToCristinRole(cristinRole);
+        return List.of(cristinRole);
+    }
+
+    private void addOrganizationInformationToCristinRole(CristinRole cristinRole) {
+        if (contributorHasValidUnitIdentifier()) {
+            CristinUnit institutionUnit = toCristinUnit(extractUnitIdentifier().orElseThrow());
+            cristinRole.setInstitutionUnit(institutionUnit);
+        } else {
+            CristinInstitution defaultOrganization = convertOrganizationToCristinInstitution(getAffiliation());
+            cristinRole.setInstitution(defaultOrganization);
+        }
+    }
+
+    private CristinRole addRolesBasedOnContributorsType(CristinRole cristinRole) {
+        if (contributorTypeCanBeMappedToCristinRole()) {
+            var cristinRoleCode = getCristinRole(getType()).orElseThrow();
+            cristinRole.setRoleCode(cristinRoleCode);
+        }
+        return cristinRole;
+    }
+
+    private boolean contributorHasValidUnitIdentifier() {
+        return extractUnitIdentifier().isPresent();
+    }
+
+    private CristinUnit toCristinUnit(String unitIdentifier) {
+        CristinUnit cristinUnit = new CristinUnit();
+        cristinUnit.setCristinUnitId(unitIdentifier);
+        return cristinUnit;
+    }
+
+    private Optional<String> extractUnitIdentifier() {
```

Comment:
method in Organization? Organization identifier is of type department

Reply:
see previous comment